openedx/core/djangoapps/credit/models.py

```diff
@@ -392,6 +396,13 @@ def get_course_requirement(cls, course_key, namespace, name):
             return None
 
 
+@receiver(models.signals.post_save, sender=CreditRequirement)
+@receiver(models.signals.post_delete, sender=CreditRequirement)
+def invalidate_credit_requirement_cache(sender, **kwargs):   # pylint: disable=unused-argument
+    """Invalidate the cache of credit requirements. """
+    RequestCache.clear_request_cache(name=CreditRequirement.CACHE_NAMESPACE)
```

Comment:
> is there a request that both reads and writes this data across multiple courses?

I was less worried about this, and more worried about a scenario where request A is reading from course A, but any request that publishes any of courses B, C, or D can invalidate our cached A entry. Thinking more about it though, I think we're fine there - the triggering model signals will never cross request boundaries either.